sample_usage/pyta_stats.py

```diff
@@ -1,62 +1,94 @@
-# Use os.walk to walk through all the files in a directory from top to bottem.
-# Everytime when os.walk walks a file, check if the file is a python file, if
-# it is, then run a check_all on it.
-# os.walk doesn't return anything
 import os
-from python_ta import *
+from os.path import join
+import python_ta
+from python_ta.reporters.stat_reporter import StatReporter
 
-
-
-multi_files = False
 # keeps track of who called stat_calculator, to tell StatReporter how to print
+multi_files = False
+# make sure these two variables get set back to [] after
+error_messages = []
+style_messages = []
 
 
 def pyta_statistics(directory):
     """
-    Recursively run python_ta.check_all() on the files in directory to the files
-    in its subdirectories.
+    Recursively run python_ta.check_all() on the files in the directory and its
+    subdirectories to collect the error and style messages.
 
     @param str directory: The string of the path way of the directory.
     @rtype: None
     """
     global multi_files
     multi_files = True
-    # walk directory, find every "file":
-    # for each .py file visited, run python_ta.check_all(reporter=StatReporter)
-    # run stats_calculator
-
-
-def stats_calculator(error_msgs, style_msgs):
+    # store information to error_messages and style_messages
+    global error_messages
+    global style_messages
+    for root_str, dir_list, file_list in os.walk(directory):
+        for file in file_list:
+            # check if file is a .py file
+            if file[-3:] == ".py":
+                python_ta.check_all(file, reporter=StatReporter)
+                # store all the msg objects of this student's files
+                error_messages.append(StatReporter._error_messages)
```

Comment:
No, @yunyiwendy, stats_calculator doesn't have access to StatReporter, StatReporter has access to stats_calculator. So you append the error_msgs and style_msgs objects that StatReporter sends to stats_calculator when the reporter calls it.